src/snovault/elasticsearch/searches/decorators.py

```diff
@@ -96,3 +96,27 @@ def wrapper(*args, **kwargs):
         return wrapper
     return decorator
 
+
+def conditional_cache(cache, condition, key):
+    '''
+    Will use cache if `condition` function is met. Generates custom key
+    using `key` function. Cache should implement mapping interface.
+    '''
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if not condition(*args, **kwargs):
+                return func(*args, **kwargs)
+            cache_key = key(*args, **kwargs)
+            try:
+                return cache[cache_key]
```

Comment:
I forgot the behavior of Redis in python, will it always raise an exception if the key is not set? I remember in Ruby it returns nil so this wouldn't work